core/trino-main/src/test/java/io/trino/sql/query/TestUnwrapCastInComparison.java

```diff
@@ -432,7 +462,7 @@ private void validate(Session session, String operator, String fromType, Object
                         "FROM (VALUES CAST(%s AS %s)) t(v)",
                 toType, operator, toValue, toType,
                 fromValue, toType, operator, toValue, toType,
-                fromValue, fromType);
+                tableValue, fromType);
```

Comment:
This is unnecessary. It can be made to support row types by changing the query to do this, instead:

```sql
FROM (VALUES CAST(ROW(%s) AS row(%s))) t(v)
```